src/mlpack/methods/ann/layer/padding_impl.hpp

```diff
@@ -34,6 +35,60 @@ PaddingType<MatType>::PaddingType(
   // Nothing to do here.
 }
 
+template<typename MatType>
+PaddingType<MatType>::PaddingType(const PaddingType& other) :
+    Layer<MatType>(other),
+    padWLeft(other.padWLeft),
+    padWRight(other.padWRight),
+    padHTop(other.padHTop),
+    padHBottom(other.padHBottom),
+    totalInMaps(other.totalInMaps)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+PaddingType<MatType>::PaddingType(PaddingType&& other) :
+    Layer<MatType>(std::move(other)),
+    padWLeft(std::move(other.padWLeft)),
+    padWRight(std::move(other.padWRight)),
+    padHTop(std::move(other.padHTop)),
+    padHBottom(std::move(other.padHBottom)),
+    totalInMaps(std::move(other.totalInMaps))
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+PaddingType<MatType>&
+PaddingType<MatType>::operator=(const PaddingType& other)
+{
+  if (this != &other)
+    Layer<MatType>::operator=(other);
+    padWLeft = other.padWLeft;
+    padWRight = other.padWRight;
+    padHTop = other.padHTop;
+    padHBottom = other.padHBottom;
+    totalInMaps = other.totalInMaps;
```

Comment:
```suggestion
  if (this != &other)
  {
    Layer<MatType>::operator=(other);
    padWLeft = other.padWLeft;
    padWRight = other.padWRight;
    padHTop = other.padHTop;
    padHBottom = other.padHBottom;
    totalInMaps = other.totalInMaps;
  }
```
Maybe too much Python? :smile: